scripts/importer/validateData.py

```diff
@@ -727,7 +727,7 @@ def checkSampleId(self, sample_id, column_number):
 
         Return True if the sample id was valid, False otherwise.
         """
-        if sample_id not in DEFINED_SAMPLE_IDS:
+        if DEFINED_SAMPLE_IDS is not None and sample_id not in DEFINED_SAMPLE_IDS:
```

Comment:
I guess these changes were needed because now an incremental import might not contain any samples. Or might not contain any patients, etc.
But in the "import whole study" mode, should we try to detect (and reject) a study which had no patients (and therefore no samples)? Or should we allow it to go in?
Maybe this is covered in the way that DEFINED_SAMPLE_IDS is initialized somewhere (None for incremental, but some empty container for whole study import)?

Reply:
In "import whole study" mode:
   - if `--relaxed_clinical_definitions` is set, `DEFINED_SAMPLE_IDS` would be equal to empty set (`set()`, not `None`) and fail the validation in this line
   - otherwise, it'd fail even earlier here https://github.com/cBioPortal/cbioportal-core/blob/rfc79/scripts/importer/validateData.py#L5430

Reply:
Ok. (although this would require the user to run the validator, unless a validator run occurs as part of scripted study import automatically)